packages/block-editor/src/components/inserter/reusable-block-list.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import { isEmpty } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { withSpokenMessages } from '@wordpress/components';
+import { useMemo, useEffect } from '@wordpress/element';
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import BlockTypesList from '../block-types-list';
+import __experimentalInserterMenuExtension from '../inserter-menu-extension';
+import { searchBlockItems } from './search-items';
+import InserterPanel from './panel';
+import InserterNoResults from './no-results';
+import useBlockTypesState from './hooks/use-block-types-state';
+
+/**
+ * List of reusable blocks shown in the "Reusable" tab of the inserter.
+ *
+ * @param {Object}   props                Component props.
+ * @param {?string}  props.rootClientId   Client id of block to insert into.
+ * @param {Function} props.onInsert       Callback to run when item is inserted.
+ * @param {Function} props.onHover        Callback to run when item is hovered.
+ * @param {?string}  props.filterValue    Search term.
+ * @param {Function} props.debouncedSpeak Debounced speak function.
+ *
+ * @return {WPComponent} The component.
+ */
+export function InserterReusableBlockList( {
+	rootClientId,
+	onInsert,
+	onHover,
+	filterValue,
+	debouncedSpeak,
+} ) {
+	const [ items, categories, collections, onSelectItem ] = useBlockTypesState(
+		rootClientId,
+		onInsert
+	);
+
+	const filteredItems = useMemo( () => {
+		return searchBlockItems(
+			items,
+			categories,
+			collections,
+			filterValue
+		).filter( ( { category } ) => category === 'reusable' );
+	}, [ filterValue, items, categories, collections ] );
+
+	// Announce search results on change.
+	useEffect( () => {
+		const resultsFoundMessage = sprintf(
+			/* translators: %d: number of results. */
+			_n( '%d result found.', '%d results found.', filteredItems.length ),
+			filteredItems.length
+		);
+		debouncedSpeak( resultsFoundMessage );
+	}, [ filterValue, debouncedSpeak ] );
+
+	const hasItems = ! isEmpty( filteredItems );
+
+	return (
+		<div>
```

Comment:
Is this container necessary?

Reply:
I'm not sure. I was just copying what `block-list.js` did.

Reply:
I've made some adjustments to clean up the DOM, so the extra `<div>`s are gone now.